service/controller/v3/resource/dnsrecord/resource.go

```diff
@@ -46,19 +49,19 @@ func (r *Resource) Name() string {
 }
 
 func (r *Resource) getDNSRecordSetsClient() (*dns.RecordSetsClient, error) {
-	azureClients, err := client.NewAzureClientSet(r.azureConfig)
+	azureClients, err := client.NewAzureClientSet(r.hostAzureConfig)
```

Comment:
I don't think it will work like that.
In current/desired we use guest client to get nameservers from guest DNS zones.
In update/delete we use host cluster client to create delegations.

Reply:
fixed